leptos_dom/src/events/typed.rs

```diff
@@ -167,6 +167,86 @@ impl DOMEventResponder for crate::View {
     }
 }
 
+/// A statically typed event handler.
+pub enum EventHandlerFn {
```

Comment:
I'm wondering if there's a way to implement this without a hardcoded list of event handlers: for example by storing the event name and a type-erased handler. The enum-based approach doesn't support custom events, and the big match statement seems like it could expensive in terms of binary size.

Reply:
Yeah, I don't particularly like this either.. I tried it a few times but couldn't find  a solution that works. The main purpose of this type is to be used in some sort of collection, so making it generic in any way should be avoided. But I do not know any trait that would be suitable used as a trait object here. Do you? Or how would you otherwise erase the function type? Maybe there is something around which I'm not aware of. Our `EventDescriptor` is only ever used in generic contexts and certainly not object safe. The `FromWasmAbi` trait also didn't led me to a solution. `gloo` (https://github.com/rustwasm/gloo/blob/master/crates/events/src/lib.rs) also has nothing of interest in this regard.

I think I would leave it at that for know. If you have any ideas on this, please let me know. Happy to change this!

Reply:
What I meant was something like this:
```rust
pub struct EventHandlerFn {
    event_name: Oco<'static, str>,
    handler: Box<dyn FnMut(web_sys::Event)>,
}

impl EventHandlerFn {
    pub fn new<E: EventDescriptor>(ev: E, handler: impl FnMut(E::EventType)) -> Self {
        Self {
            event_name: ev.name(),
            handler: Box::new(move |ev: Event| {
                // convert `Event` into `E::EventType` using JsCast
            }),
        }
    }
}
```
Unfortunately based on the way I defined `EventDescriptor` at some point, I don't think we can actually convert an `E::EventType` into a `web_sys::Event`, even using `.unchecked_into()` — even though all of these types actually inherit from Event and that conversion should be fine.

For now then I think your solution is probably best!